### PR TITLE
fix: only allow display flex in media query build project landing page project

### LINKED ISSUE #46887: Product Landing Page: allow flexbox to be set only inside a media query
### Describe the Issue

The test should allow that flexbox is only set inside a media query.

As the test is written right now it doesn't seem possible.

I didn't actually try logging out what the test is accessing. But I would assume from looking at the code that if the selector is inside a media query `rule.style` won't be what is expected.

https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/14-responsive-web-design-22/build-a-product-landing-page-project/build-a-product-landing-page.md

```js
const stylesheet = new __helpers.CSSHelp(document).getStyleSheet()
const cssRules = new __helpers.CSSHelp(document).styleSheetToCssRulesArray(stylesheet)
const usesFlex = cssRules.find(rule => {
  return rule.style?.display === 'flex' || rule.style?.display === 'inline-flex'
})
assert(usesFlex)
```

Forum: https://forum.freecodecamp.org/t/product-landing-page-build-a-product-landing-page-8w6hc8gqzk1w-bqdykvdf/528722

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/build-a-product-landing-page-project/build-a-product-landing-page

### Your code

```

```

### Expected behavior

Allow flexbox to be set inside a media query.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This gives people the freedom to still add `display: flex` but adding it to the mediaquery mandatory. 

Closes #46887

### CODE REVIEW

**Review by hanswang123456 — COMMENTED** [review]
This does resolve the issue, especially with the updated description.

**Review by moT01 — CHANGES_REQUESTED** [review]
I don't think we should require the property to be in a media query. This would break already completed projects that did not put it there. If we can, we should stick with current test (text) that just checks if the project uses the property at least once - and adjust the test so it can be in, or not in, a media query.

Also, when we get this figured out - we want to mimic the changes in the duplicate legacy RWD project.

File: curriculum/challenges/english/14-responsive-web-design-22/build-a-product-landing-page-project/build-a-product-landing-page.md [hunk 1]
```diff
@@ -251,15 +251,21 @@ const cssCheck = new __helpers.CSSHelp(document).getCSSRules('media')
 assert(cssCheck.length > 0 || htmlSourceAttr.length > 0);
 ```
 
-Your Product Landing Page should use CSS Flexbox at least once.
+Your media query should contain a `display` property with the value set to `flex` or `inline-flex`
```
Comment (hanswang123456) [inline curriculum/challenges/english/14-responsive-web-design-22/build-a-product-landing-page-project/build-a-product-landing-page.md:254]: I think giving such detailed instruction isn't necessary. Just let the user know they need to use flexbox.

**Comment by hanswang123456** [comment]
> I don't think we should require the property to be in a media query. This would break already completed projects that did not put it there. If we can, we should stick with current test (text) that just checks if the project uses the property at least once - and adjust the test so it can be in, or not in, a media query.
> 
> Also, when we get this figured out - we want to mimic the changes in the duplicate legacy RWD project.

Perhaps keep the old code and add on the checks for media query CSS

**Pull request closed without merging**